docs/build.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+
+import importlib
+import logging
+import sys
+from pathlib import Path
+
+import pydantic
+
+logger = logging.getLogger(__file__)
+ROOT = Path(__file__).parent.parent
+JSON_SCHEMAS_ROOT = ROOT / "docs" / "json_schemas"
+
+
+def build_schema(schema_path: Path, interface_name: str, version:int,
```

Comment:
Please include tests for those new methods.

Reply:
This method is pretty long and complicated, I suspect writing tests will also put pressure into refactoring/simplifying it

Reply:
Same goes for linting, we should follow same practices as in other Python projects here. As it is right now `pflake8` fails at a couple of places

Reply:
on it

Reply:
projectified the whole thing